core/src/main/scala/com/softwaremill/react/kafka2/Consumer.scala

```diff
@@ -75,7 +89,7 @@ object ManualCommitConsumer {
   }
 }
 
-class ManualCommitConsumer[K, V](consumerProvider: ConsumerProvider[K, V])
+class ManualCommitConsumer[K, V](consumerProvider: () => KafkaConsumer[K, V])
```

Comment:
The settings could just carry the offset and partitions information and we call the methods on the `KafkaConsumer` when we create it. The settings doesn't have to be only backed by a `Map`. Some of these settings should perhaps also be populated from config.

perhaps we can keep this for our own testing purposes, but not expose it in end user api?